### DISCUSSION TITLE
Webpack dependency: 4 vs. 5

### DISCUSSION THREAD

**Discussion by wol81** [vuejs/vue-cli#6650] [Q&A]
I have added "@vue/cli-service": "~5.0.0-beta.3" to the package.json of my typescript project.

That one has (among others) dependencies to
- "webpack": "^5.22.0",
- "@types/webpack-dev-server": "^3.11.0",

According to "npm list", the latter one depends on @types/webpack@4.41.30, and also in node_modules/@types/webpack-dev-server/package.json, I find a dependency to "@types/webpack": "^4",

So, I get webpack 5 via dependency, and webpack 4 via the other chain.

I suspect that this might cause me having that problem: https://github.com/webpack/webpack/issues/12185
Especially that comment: https://github.com/webpack/webpack/issues/12185#issuecomment-759136246

I must admit that I am relatively new to that infrastructure, so I might easily miss something fundamental...

Nevertheless question: Is that dependency situation intended?

**Comment by haoqunjiang** [comment]
Yeah, it's expected. `webpack-dev-server` v4 was released very recently. I'm trying to upgrade it in the next beta.
